### PR TITLE
chore(ccipdata): split ExecOffchainConfig definitions

### PR DESCRIPTION
**Pull request by roman-kashitsyn**
## Motivation

Currently, we use the `ExecOffchainConfig` struct in two contexts:
* As the expected configuration format that we decode from the plugin configuration.
* As an internal interface type between the offramp reader interface and the execution reporting plugin.

So far, that has worked well, but this coupling hinders our ability to evolve the configuration format in v1.3 without breaking compatibility with older versions.

## Solution

This change splits the `ExecOffchainConfig` struct into two types:
 * A version-dependent struct we parse from JSON (defined in `v1_0_0/offramp.go`)
 * A version-independent interface type defined in offramp_reader.go.

It also documents the fields in the structs.

This change has no functional changes. A future change will implement a v1.3-specific offramp reader using the new configuration struct.

### CODE REVIEW

File: core/services/ocr2/plugins/ccip/internal/ccipdata/offramp_reader.go [hunk 1]
```diff
@@ -19,45 +19,20 @@ const (
 	ManuallyExecute = "manuallyExecute"
 )
 
-// ExecOffchainConfig Do not change the JSON format of this struct without consulting with the RDD people first.
+// An ExecOffchainConfig specifies configuration for nodes executing committed messages.
 type ExecOffchainConfig struct {
```
Comment (makramkd) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/offramp_reader.go]: Nit: Go convention/doc comment rules dictate that comments start with the component name, so

```suggestion
// ExecOffchainConfig specifies configuration for nodes executing committed messages.
type ExecOffchainConfig struct {
```

File: core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go [hunk 2]
```diff
@@ -92,6 +93,57 @@ func (d ExecOnchainConfig) PermissionLessExecutionThresholdDuration() time.Durat
 	return time.Duration(d.PermissionLessExecutionThresholdSeconds) * time.Second
 }
 
+// An ExecOffchainConfig is the configuration for nodes executing committed CCIP messages (v1.0–v1.2).
+// It comes from the OffchainConfig field of the corresponding OCR2 plugin configuration.
+// NOTE: do not change the JSON format of this struct without consulting with the RDD people first.
+type ExecOffchainConfig struct {
```
Comment (makramkd) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go]: Ditto here re: comment starting word.

File: core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go [hunk 3]
```diff
@@ -92,6 +93,57 @@ func (d ExecOnchainConfig) PermissionLessExecutionThresholdDuration() time.Durat
...
+	// SourceFinalityDepth indicates how many confirmations a transaction should get on the source chain event before we consider it finalized.
+	SourceFinalityDepth uint32
+	// See [ccipdata.ExecOffchainConfig.DestOptimisticConfirmations]
+	DestOptimisticConfirmations uint32
+	// SourceFinalityDepth indicates how many confirmations a transaction should get on the destination chain event before we consider it finalized.
```
Comment (makramkd) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/v1_0_0/offramp.go]: Typo: `DestFinalityDepth`
```suggestion
	// DestFinalityDepth indicates how many confirmations a transaction should get on the destination chain event before we consider it finalized.
```

**Review by makramkd — COMMENTED** [review]
Looking good, minor comments!

File: core/services/ocr2/plugins/ccip/internal/ccipdata/offramp_reader.go (hunk 1 above)
Comment (roman-kashitsyn) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/offramp_reader.go]: I followed the [Go Doc recommendation for types](https://tip.golang.org/doc/comment#type):

> A type’s doc comment should explain what each instance of that type represents or provides.
> `// A Reader serves content from a ZIP archive.`

I can remove the indefinite article if it's more consistent with the rest of the code base.

Comment (makramkd) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/offramp_reader.go]: Oh, looks like articles could be present or absent! Didn't know that. I think it's fine, as long as it matches the Go Doc rules 😄

**Comment by roman-kashitsyn** [comment]
> Looking good, minor comments!

@makramkd, I've added a test to make `codecov/patch` happy. Could you please take a quick look and approve if there are no more comments?

File: core/services/ocr2/plugins/ccip/internal/ccipdata/offramp_reader_test.go [hunk 4]
```diff
@@ -55,13 +56,13 @@ func TestOffRampFilters(t *testing.T) {
 	}, 3)
 }
 
-func TestExecOffchainConfig_Encoding(t *testing.T) {
+func TestExecOffchainConfig100_Encoding(t *testing.T) {
```
Comment (jarnaud) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/offramp_reader_test.go]: Could we move this test under `/v1_0_0` if it's version specific?

Comment (roman-kashitsyn) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/offramp_reader_test.go]: Done.

**Review by makramkd — APPROVED** [review]

**Pull request merged**